ui/ducks/bridge/actions.ts

```diff
@@ -70,3 +104,356 @@ export const setToChain = (chainId: Hex) => {
     );
   };
 };
+
+export const getBridgeERC20Allowance = async (
+  contractAddress: string,
+  walletAddress: string,
+  chainId: Hex,
+): ReturnType<BridgeController['getBridgeERC20Allowance']> => {
+  return await submitRequestToBackground(
+    BridgeBackgroundAction.GET_BRIDGE_ERC20_ALLOWANCE,
+    [contractAddress, walletAddress, chainId],
+  );
+};
+
+export const submitBridgeTransaction = (
+  history: ReturnType<typeof useHistory>,
+) => {
+  return async (
+    dispatch: MetaMaskReduxDispatch,
+    getState: () => MetaMaskReduxState & BridgeAppState,
+  ) => {
+    const state = getState();
+    const isBridgeEnabled = getIsBridgeEnabled(state);
+    const isBridgeChain = getIsBridgeChain(state);
+
+    if (!(isBridgeEnabled && isBridgeChain)) {
+      // TODO do we want to do something here?
+      return;
+    }
+
+    const quoteMetas = getQuotes(state);
+    const quoteMeta = quoteMetas[0];
```

Comment:
What do you think of passing `quoteMeta` in as a param instead of querying state? In the [quote-sorting PR](https://github.com/MetaMask/metamask-extension/pull/27731) I plan to create a hook that calculates quote return values and provides the recommended/selected quote

Reply:
That's fine with me, it was all temporary until I linked up the quotes + the tx submission PRs

Reply:
Done in [9c4ed4d](https://github.com/MetaMask/metamask-extension/pull/27262/commits/9c4ed4d240e19d5723a4483d9a39ef5c669ea711)